test/e2e/metamask-ui.spec.js

```diff
@@ -325,145 +299,127 @@ describe('MetaMask', function () {
       await driver.delay(regularDelayMs);
 
       // Continue to next screen
-      await driver.clickElement(By.xpath(`//button[contains(text(), 'Next')]`));
+      await driver.clickElement({ text: 'Next', tag: 'button' });
       await driver.delay(regularDelayMs);
     });
 
     it('confirms the transaction', async function () {
-      await driver.clickElement(
-        By.xpath(`//button[contains(text(), 'Confirm')]`),
-      );
+      await driver.clickElement({ text: 'Confirm', tag: 'button' });
       await driver.delay(largeDelayMs * 2);
     });
 
     it('finds the transaction in the transactions list', async function () {
-      await driver.clickElement(By.css('[data-testid="home__activity-tab"]'));
+      await driver.clickElement('[data-testid="home__activity-tab"]');
       await driver.wait(async () => {
         const confirmedTxes = await driver.findElements(
-          By.css(
-            '.transaction-list__completed-transactions .transaction-list-item',
-          ),
+          '.transaction-list__completed-transactions .transaction-list-item',
         );
         return confirmedTxes.length === 1;
       }, 10000);
 
       const txValues = await driver.findElement(
-        By.css('.transaction-list-item__primary-currency'),
+        '.transaction-list-item__primary-currency',
       );
       await driver.wait(until.elementTextMatches(txValues, /-1\s*ETH/u), 10000);
     });
   });
 
   describe('Send ETH from inside MetaMask using fast gas option', function () {
     it('starts a send transaction', async function () {
-      await driver.clickElement(By.css('[data-testid="eth-overview-send"]'));
+      await driver.clickElement('[data-testid="eth-overview-send"]');
       await driver.delay(regularDelayMs);
 
       const inputAddress = await driver.findElement(
-        By.css('input[placeholder="Search, public address (0x), or ENS"]'),
+        'input[placeholder="Search, public address (0x), or ENS"]',
       );
       await inputAddress.sendKeys('0x2f318C334780961FB129D2a6c30D0763d9a5C970');
 
-      const inputAmount = await driver.findElement(
-        By.css('.unit-input__input'),
-      );
+      const inputAmount = await driver.findElement('.unit-input__input');
       await inputAmount.sendKeys('1');
 
       const inputValue = await inputAmount.getAttribute('value');
       assert.equal(inputValue, '1');
 
       // Set the gas price
-      await driver.clickElement(
-        By.xpath(`//button/div/div[contains(text(), "Fast")]`),
-      );
+      await driver.clickElement({ text: 'Fast', tag: 'button/div/div' });
```

Comment:
I guess this means the `tag` property is only compatible with xpath :thinking: 

It might be worth using the `xpath` locator here explicitly if we want to make the `text` and `tag`-based locators implementable without using xpath.

Reply:
How would you make text locators implementable without xpath with Selenium? For what its worth this can be expressed very easily with Playwright `button > div > div:has-text("fast")`

Reply:
Oh I meant in Playwright - I think we need to use xpath for this in Selenium.

if this slash syntax is easily parsed/handled by Playwright that works too! Just wasn't sure it was portable. That approach you just gave an example of seems fine to me.